utilities/resources.py

```
@@ -70,6 +70,27 @@ def create_and_store_resource(
+    Args:
+        fixture_store (dict[str, Any]): Fixture store for resource tracking.
+        kind (str): Resource kind key in fixture_store["teardown"].
+        name (str): Resource name to unregister.
+
+    Raises:
+        ValueError: If no matching resource entry is found for kind/name.
+    """
+    resources = fixture_store["teardown"][kind]
+    remaining = [resource for resource in resources if resource["name"] != name]
+    if len(remaining) == len(resources):
+        raise ValueError(f"Resource '{name}' of kind '{kind}' not found in fixture_store teardown")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)